src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -141,5 +142,43 @@ def _produce_billing_outcome(
             quantity=quantity,
         )
 
+    def _flag_metric_received_for_project(self, generic_metric: GenericMetric) -> None:
+        try:
+            org_id = generic_metric["org_id"]
+            project_id = generic_metric["project_id"]
+
+            cache_key = _get_project_flag_updated_cache_key(org_id, project_id)
+            if cache.get(cache_key) is None:
```

Comment:
How does this cache key get set so that the rest of the function runs? The only write I can see to this key is done after this check. Perhaps this needs to be `is not`?

Reply:
Yep sorry, in the inlining of the functions I put the wrong condition. I will refine everything now, thanks for checking!